app/experimenter/nimbus-ui/src/components/ChangeApprovalOperations/index.test.tsx

```diff
@@ -0,0 +1,207 @@
+  });
+
+  it("when user cannot review and review has been approved, an approval pending notice is displayed", async () => {
+    const startRemoteSettingsApproval = jest.fn();
+    render(
+      <Subject
+        {...{
+          canReview: false,
+          reviewRequestEvent: mockChangelog(),
+          approvalEvent: mockChangelog(),
+          startRemoteSettingsApproval,
+        }}
+      />,
+    );
+    await screen.findByTestId("approval-pending");
+    expect(
+      screen.queryByTestId("open-remote-settings"),
+    ).not.toBeInTheDocument();
+  });
+
+  it("when user can review, supports rejection and supplying a reason", async () => {
+    const expectedReason = "hovercraft contains eels";
+    const rejectChange = jest.fn();
+    render(
+      <Subject
+        {...{
+          canReview: true,
+          reviewRequestEvent: mockChangelog(),
+          rejectChange,
+        }}
+      />,
+    );
+    const rejectButton = await screen.findByTestId("reject-request");
+    await act(async () => void fireEvent.click(rejectButton));
+    const rejectSubmitButton = await screen.findByTestId("reject-submit");
+    const rejectReasonField = await screen.findByTestId("reject-reason");
+    expect(rejectSubmitButton).not.toBeEnabled();
+    await act(async () => {
+      fireEvent.change(rejectReasonField, {
+        target: { value: expectedReason },
+      });
+      fireEvent.blur(rejectReasonField);
+    });
+    expect(rejectSubmitButton).toBeEnabled();
+    await act(async () => void fireEvent.click(rejectSubmitButton));
+    expect(rejectChange).toBeCalledWith({ reason: expectedReason });
+  });
+
+  it("when user can review, supports rejection and cancelling", async () => {
+    const rejectChange = jest.fn();
+    render(
+      <Subject
+        {...{
+          canReview: true,
+          reviewRequestEvent: mockChangelog(),
+          rejectChange,
+        }}
+      />,
+    );
+    const rejectButton = await screen.findByTestId("reject-request");
+    await act(async () => void fireEvent.click(rejectButton));
+    const rejectCancelButton = await screen.findByTestId("reject-cancel");
+    await act(async () => void fireEvent.click(rejectCancelButton));
+    await screen.findByTestId("approve-request");
+    await screen.findByTestId("reject-request");
+    expect(screen.queryByTestId("reject-reason")).not.toBeInTheDocument();
+    expect(rejectChange).not.toHaveBeenCalled();
+  });
+
+  it("reports a rejection reason when review is rejected", async () => {
+    const actionDescription = "gizmofy";
+    const rejectionUser = "jdoe@mozilla.com";
+    const rejectionMessage = "chutes too narrow";
+    render(
+      <Subject
+        {...{
+          actionDescription,
+          reviewRequestEvent: mockChangelog(),
+          rejectionEvent: mockChangelog(rejectionUser, rejectionMessage),
+        }}
+      />,
+    );
+    await screen.findByTestId("action-button");
+    await screen.findByTestId("rejection-notice");
```

Comment:
I think this is testing the same thing as L169 below, just differently?

Reply:
Yeah, pretty much. I think I wanted to be sure both the vaguely correct message appeared, as well as the parent container. Not really concise